3rdparty/openvx/include/openvx_hal.hpp

```diff
@@ -184,6 +269,444 @@ inline int ovx_hal_add(const T *a, size_t astep, const T *b, size_t bstep, T *c,
     return CV_HAL_ERROR_OK;
 }
 
+#if defined OPENCV_IMGPROC_HAL_INTERFACE_H
+#define CV_HAL_INTER_NEAREST 0
+#define CV_HAL_INTER_LINEAR 1
+#define CV_HAL_INTER_CUBIC 2
+#define CV_HAL_INTER_AREA 3
+#define CV_HAL_INTER_LANCZOS4 4
+#define MORPH_ERODE 0
+#define MORPH_DILATE 1
+
+inline int ovx_hal_resize(int atype, const uchar *a, size_t astep, int aw, int ah, uchar *b, size_t bstep, int bw, int bh, double inv_scale_x, double inv_scale_y, int interpolation)
+{
+    try
+    {
+        vxContext * ctx = vxContext::getContext();
+        vxImage ia(*ctx, a, astep, aw, ah);
+        vxImage ib(*ctx, b, bstep, bw, bh);
+
+        if(!((atype == CV_8UC1 || atype == CV_8SC1) &&
+             inv_scale_x > 0 && inv_scale_y > 0 &&
+             (bw - 0.5) / inv_scale_x - 0.5 < aw && (bh - 0.5) / inv_scale_y - 0.5 < ah &&
+             (bw + 0.5) / inv_scale_x + 0.5 >= aw && (bh + 0.5) / inv_scale_y + 0.5 >= ah &&
+             std::abs(bw / inv_scale_x - aw) < 0.1 && std::abs(bh / inv_scale_y - ah) < 0.1 ))
+            vxErr(VX_ERROR_INVALID_PARAMETERS, "Bad scale").check();
```

Comment:
Probably we should not return error in such cases, but `CV_HAL_ERROR_NOT_IMPLEMENTED`. Many tests are failing because of this.